src/apify/_memory_storage/resource_clients/dataset.py

```diff
@@ -94,36 +97,36 @@ async def update(self, *, name: Optional[str] = None) -> Dict:
         )
 
         if existing_dataset_by_id is None:
-            _raise_on_non_existing_storage(_StorageTypes.DATASET, self._id)
+            raise_on_non_existing_storage(StorageTypes.DATASET, self._id)
 
         # Skip if no changes
         if name is None:
             return existing_dataset_by_id._to_resource_info()
 
-        async with existing_dataset_by_id._file_operation_lock:
+        async with existing_dataset_by_id._file_operation_lock:  # type: ignore
             # Check that name is not in use already
             existing_dataset_by_name = next(
                 (dataset for dataset in self._memory_storage_client._datasets_handled if dataset._name and dataset._name.lower() == name.lower()),
                 None,
             )
 
             if existing_dataset_by_name is not None:
-                _raise_on_duplicate_storage(_StorageTypes.DATASET, 'name', name)
+                raise_on_duplicate_storage(StorageTypes.DATASET, 'name', name)
 
             existing_dataset_by_id._name = name
 
             previous_dir = existing_dataset_by_id._resource_directory
 
             existing_dataset_by_id._resource_directory = os.path.join(self._memory_storage_client._datasets_directory, name)
 
-            await _force_rename(previous_dir, existing_dataset_by_id._resource_directory)
+            await force_rename(previous_dir, existing_dataset_by_id._resource_directory)
 
             # Update timestamps
-            await existing_dataset_by_id._update_timestamps(True)
+            await existing_dataset_by_id._update_timestamps(has_been_modified=True)  # type: ignore
```

Comment:
why is the type ignore needed now?

Reply:
The problem with all of the new `# type: ignore` in `resource_clients/` is the following...

The `_find_or_create_client_by_id_or_name` method returns either `BaseResourceClient` or `None`.

![image](https://github.com/apify/apify-sdk-python/assets/25082181/422d775a-8644-4df8-87b5-080e0b4ad39f)

There is a check for a `None` value, so after that, the IDE expects it to be `BaseResourceClient`.

However, `BaseResourceClient` does not have all of the fields that are accessed here (`_file_operation_lock`, `_update_timestamps`, ...). Only its child/children have it. 

![image](https://github.com/apify/apify-sdk-python/assets/25082181/7fd8da34-135c-4ddb-99dd-27d2bcee9968)

So that's the problem and probably always has been. The reason why it was not caught before, might be the update of Mypy from 1.5 to 1.7?

Reply:
Ok, I created a new issue for that - https://github.com/apify/apify-sdk-python/issues/152, resolving.